python/tests/catalog/test_rest.py

```diff
@@ -0,0 +1,670 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from uuid import UUID
+
+import pytest
+import requests_mock
+
+from pyiceberg.catalog.base import PropertiesUpdateSummary
+from pyiceberg.catalog.rest import RestCatalog, RestTable, TokenResponse
+from pyiceberg.exceptions import (
+    AlreadyExistsError,
+    BadCredentialsError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+    TableAlreadyExistsError,
+)
+from pyiceberg.schema import Schema
+from pyiceberg.table.base import Table
+from pyiceberg.table.metadata import TableMetadataV1
+from pyiceberg.table.partitioning import PartitionField, PartitionSpec
+from pyiceberg.table.refs import SnapshotRef, SnapshotRefType
+from pyiceberg.table.snapshots import Operation, Snapshot, Summary
+from pyiceberg.table.sorting import SortField, SortOrder
+from pyiceberg.transforms import IdentityTransform, TruncateTransform
+from pyiceberg.types import (
+    BooleanType,
+    IntegerType,
+    NestedField,
+    StringType,
+)
+
+TEST_HOST = "https://iceberg-test-catalog/"
+TEST_TOKEN = "client:secret"
+
+
+def test_token_200():
+    with requests_mock.Mocker() as m:
+        m.post(
+            f"{TEST_HOST}oauth/tokens",
+            json={
+                "access_token": TEST_TOKEN,
+                "token_type": "Bearer",
+                "expires_in": 86400,
+                "warehouse_id": "8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+                "region": "us-west-2",
```

Comment:
`warehouse_id` and `region` aren't in the spec and should probably be omitted in these tests.

Reply:
And they are gone!